Fix deleting loadbalancer after resource group in azure

phiphi282 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:

When trying to delete a loadbalancer after all resources in azure
including the resource group have been deleted we currently get an an
error when trying to delete the loadbalancer afterwards because the
resource group hasn't been found.

Change EnsureLoadBalancerDeleted function to not fail when the resource
group has already been deleted.

The bug was initially fixed in #75256 but it looks like it got removed in #86555
Initial issue was #75198

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #75198

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Fixed a regression in 1.18+ where loadbalancer deletion gets stuck because of missing resource group #75198
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

phiphi282 commented:
/assign @andyzhangx

andyzhangx commented:
/ok-to-test
thanks for the contribution. Could you squash your commits first? thanks.

phiphi282 committed:
Allow 404 error on lb deletion in azure

When trying to delete a loadbalancer after all resources in azure
including the resource group have been deleted we currently get an an
error when trying to delete the loadbalancer afterwards because the
resource group hasn't been found.

Change EnsureLoadBalancerDeleted function to not fail when the resource
group has already been deleted.

phiphi282 commented:
/retest

phiphi282 commented:
The [pull-kubernetes-e2e-azure-disk](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93962/pull-kubernetes-e2e-azure-disk/1294017838159237120) job is currently stuck after failing some commands with `no space left on device`. I am not sure how to abort or restart the job currently. The documentation states that `/restart` only restarts failed jobs.

I guess I should probably create an issue for this but I am not sure if this is a bug or a flaky test.

andyzhangx commented:
/retest pull-kubernetes-e2e-azure-disk

andyzhangx commented:
/test pull-kubernetes-e2e-azure-disk

andyzhangx reviewed (approved):
/kind bug
/priority important-soon
/sig cloud-provider
/area provider/azure
/lgtm

would you take a look? @feiskyer
looks like need to cherry-pick this PR to old release

feiskyer reviewed (commented):
thanks for the fix

/lgtm
/approve

phiphi282 commented:
So i looked at the code for older releases and the bug only exists from v1.18 onwards. So the pr only needs to be cherry-picked for 1.18.

craiglpeters commented:
/milestone v1.20

liggitt commented:
this will be moved back into the milestone as part of the [phased reopen for 1.20 merges](https://docs.google.com/document/d/1yI7o7R2bOB7QtkHUFMHMb5XObiJmzIAnJPuxhdAX-CA/edit)

phiphi282 commented:
/retest

phiphi282 commented:
If test will fail again I will rebase master and hope that tests have become less flaky :)

phiphi282 commented:
/retest

phiphi282 commented:
/retest

feiskyer reviewed (commented):
/lgtm

phiphi282 commented:
I think this still needs to be cherry-picked for 1.18 and 1.19.
Is there something I need to/can to for this? :)

andyzhangx commented:
> I think this still needs to be cherry-picked for 1.18 and 1.19.
> Is there something I need to/can to for this? :)

@phiphi282 i have already cherry-picked the PRs, PTAL, and thanks for the contribution!
